crates/l2/common/src/prover.rs

```diff
@@ -70,20 +70,75 @@ impl ProverType {
                 vec![Value::Bytes(vec![].into()), Value::Bytes(vec![].into())]
             }
             ProverType::Exec => unimplemented!("Doesn't need to generate an empty calldata."),
-            ProverType::Aligned => unimplemented!("Doesn't need to generate an empty calldata."),
         }
     }
 
+    /// Used to call a getter for the REQUIRE_*_PROOF boolean in the OnChainProposer contract
     pub fn verifier_getter(&self) -> Option<String> {
         // These values have to match with the OnChainProposer.sol contract
         match self {
-            Self::Aligned => Some("ALIGNEDPROOFAGGREGATOR()".to_string()),
-            Self::RISC0 => Some("R0VERIFIER()".to_string()),
-            Self::SP1 => Some("SP1VERIFIER()".to_string()),
-            Self::TDX => Some("TDXVERIFIER()".to_string()),
+            Self::RISC0 => Some("REQUIRE_RISC0_PROOF()".to_string()),
+            Self::SP1 => Some("REQUIRE_SP1_PROOF()".to_string()),
+            Self::TDX => Some("REQUIRE_TDX_PROOF()".to_string()),
             Self::Exec => None,
         }
     }
+
+    /// Fills the "vm_program_code" field of an AlignedVerificationData struct,
+    /// used for sending a proof to Aligned Layer.
+    pub fn aligned_vm_program_code(&self) -> std::io::Result<Option<Vec<u8>>> {
+        match self {
+            Self::RISC0 => {
+                let path = format!(
```

Comment:
This won't work on downloaded binaries. We should embed this in the binary. For reference, here's a way to do it:
https://github.com/lambdaclass/ethrex/pull/4014/files#diff-a73d7b3e1eec2d556992a25ceef5203288ab90a0f9bcc7b4a176a754c647dc42R72